## :pill: CircleCI build failures summary and remediations
As of commit c6a7672:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=c6a7672dab5af4c7697339d6c09f39197826f9fd).

### :detective: 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4007736) pytorch_xla_linux_xenial_py3_6_clang7_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=76916534) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=76916534))
<details>
<summary>
<code>Dec 18 21:41:19 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-serialization.diff</code>
</summary>

```
Dec 18 21:41:19 + python /var/lib/jenkins/workspace/xla/scripts/cond_patch.py /var/lib/jenkins/workspace/xla/scripts/../torch_patches /var/lib/jenkins/workspace/xla/scripts/../.. 
Dec 18 21:41:19 1 out of 1 hunk FAILED 
Dec 18 21:41:19 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Dec 18 21:41:19 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-device_test.diff 
Dec 18 21:41:19 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-serialization.diff 
Dec 18 21:41:19 Traceback (most recent call last): 
Dec 18 21:41:19   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 67, in <module> 
Dec 18 21:41:19     patch_repo(args) 
Dec 18 21:41:19   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 49, in patch_repo 
Dec 18 21:41:19     raise RuntimeError('Failed to apply patch: {}'.format(ppath)) 
Dec 18 21:41:19 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-serialization.diff 
Dec 18 21:41:19 + cleanup 
Dec 18 21:41:19 + retcode=1 
Dec 18 21:41:19 + set +x 
Dec 18 21:41:19 =================== sccache compilation log =================== 
Dec 18 21:41:19 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Dec 18 21:41:19 Compile requests                 1 
Dec 18 21:41:19 Compile requests executed        0 
Dec 18 21:41:19 Cache hits                       0 
Dec 18 21:41:19 Cache misses                     0 
Dec 18 21:41:19 Cache timeouts                   0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

--- a/test/backward_compatibility/check_backward_compatibility.py
+++ b/test/backward_compatibility/check_backward_compatibility.py
@@ -16,19 +16,7 @@ from torch._C import parse_schema
 #     believe you can land your diff before then.
 #
 # Whitelist entries can be removed after the date listed on them passes.
-white_list = [
-    ('c10_experimental', datetime.date(2020, 1, 1)),
-    ('_batch_norm_impl_index', datetime.date(2019, 11, 15)),
-    ('_batch_norm_impl_index_backward', datetime.date(2019, 11, 15)),
-    ('cudnn_batch_norm', datetime.date(2019, 11, 15)),
-    ('cudnn_batch_norm_backward', datetime.date(2019, 11, 15)),
-    ('_nnpack_spatial_convolution', datetime.date(2019, 11, 12)),
-    ('_aten', datetime.date(2019, 12, 22)),
-    ('_prim::ListConstruct', datetime.date(2019, 11, 22)),
-    ('thnn_conv3d', datetime.date(9999, 1, 1)),
-    ('thnn_conv3d.out', datetime.date(9999, 1, 1)),
-    ('grad', datetime.date(2020, 1, 1)),
-]
+white_list = []
 
 
 def white_listed(schema, white_list):
@@ -43,6 +31,8 @@ def white_listed(schema, white_list):
 
 def check_bc(new_schema_dict):
     existing_schemas = torch._C._jit_get_all_schemas()
+    is_bc = True
+    broken_ops = []
     for existing_schema in existing_schemas:
         if white_listed(existing_schema, white_list):
             print("skipping schema: ", str(existing_schema))
@@ -60,13 +50,17 @@ def check_bc(new_schema_dict):
                   .format(
                       str(existing_schema),
                       "\n\t".join(str(s) for s in new_schemas)))
-            print('The PR is introducing backward incompatible changes to the '
-                  'operator library. Please contact PyTorch team to confirm '
-                  'whether this change is wanted or not.')
             # TODO Print out more details about why candidates don't match.
-            return False
-    print('Found backward compatible schemas for all existing schemas')
-    return True
+            broken_ops.append(str(existing_schema))
+            is_bc = False
+    if is_bc:
+        print('Found backward compatible schemas for all existing schemas')
+    else:
+        print('The PR is introducing backward incompatible changes to the '
+              'operator library. Please contact PyTorch team to confirm '
+              'whether this change is wanted or not. \n Broken ops: [\n{}]'
+              .format("\n".join(broken_ops)))
+    return is_bc
 
 
 if __name__ == '__main__':

--- a/test/backward_compatibility/check_backward_compatibility.py
+++ b/test/backward_compatibility/check_backward_compatibility.py
@@ -16,7 +16,25 @@ from torch._C import parse_schema
 #     believe you can land your diff before then.
 #
 # Whitelist entries can be removed after the date listed on them passes.
-white_list = []
+white_list = [
+    ("aten::append", datetime.date(9999, 1, 1)),
+    ("prim::AutogradAnyNonZero", datetime.date(9999, 1, 1)),
+    ("aten::grad", datetime.date(9999, 1, 1)),
+    ("_c10_experimental", datetime.date(9999, 1, 1)),
+    ("aten::thnn_conv3d", datetime.date(9999, 1, 1)),
+    ("aten::native_layer_norm_double_backward", datetime.date(9999, 1, 1)),
+    ("aten::cudnn_batch_norm", datetime.date(9999, 1, 1)),
+    ("aten::cudnn_batch_norm_backward", datetime.date(9999, 1, 1)),
+    ("aten::_batch_norm_impl_index_backward", datetime.date(9999, 1, 1)),
+    ("aten::empty_like", datetime.date(9999, 1, 1)),
+    ("aten::_batch_norm_impl_index", datetime.date(9999, 1, 1)),
+    ("aten::index_fill_", datetime.date(9999, 1, 1)),
+    ("aten::index_fill", datetime.date(9999, 1, 1)),
+    ("aten::log_softmax", datetime.date(9999, 1, 1)),
+    ("aten::softmax", datetime.date(9999, 1, 1)),
+    ("aten::thnn_conv3d_forward", datetime.date(9999, 1, 1)),
+    ("aten::thnn_conv3d_backward.output_mask", datetime.date(9999, 1, 1)),
+]
 
 
 def white_listed(schema, white_list):

